bigbluebutton-html5/imports/ui/components/audio/audio-modal/component.jsx

```diff
@@ -162,23 +165,15 @@ class AudioModal extends Component {
 
   componentDidMount() {
     const {
-      joinFullAudioImmediately,
-      joinFullAudioEchoTest,
       forceListenOnlyAttendee,
       audioLocked,
+      joinFullAudioImmediately,
+      joinFullAudioEchoTest,
     } = this.props;
 
-    if (joinFullAudioImmediately) {
-      this.handleJoinMicrophone();
-    }
-
-    if (joinFullAudioEchoTest) {
-      this.handleGoToEchoTest();
-    }
-
-    if (forceListenOnlyAttendee || audioLocked) {
-      this.handleJoinListenOnly();
-    }
+    if (forceListenOnlyAttendee) return this.handleJoinListenOnly();
```

Comment:
Seems we did not recover the `audioLocked` case - check the original case. This would be part of Lock settings - joining audio as viewer with mic locked but no necessarily `forceListenOnlyAttendee` enabled